### ISSUE TITLE
Build a pledge form

### ISSUE THREAD

**Issue by boulderalf** [BeeSafeEarth/BeeSafeEarth#6]
<a href="https://github.com/boulderalf"><img src="https://avatars.githubusercontent.com/u/2962724?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [boulderalf](https://github.com/boulderalf)**
_Monday Mar 21, 2016 at 22:35 GMT_
_Originally opened as https://github.com/boulderalf/BeeSafeEarth/issues/6_

---

Design and build a pledge form that receives information like the Google Form in http://beesafeboulder.org/pledge   The pledge form does not need to be limited by the look and feel of the Google Form.  Please share design or mockup before embarking on full development.

**Comment by ZebGirouard** [comment]
@boulderalf , @kristinreynolds , @DaveLaskarzewski , @Beesafedavid , I made a rough draft on SquareSpace today...not sure who has access to it, but I can send screenshots if requested:

https://kristin-reynolds-o4dy.squarespace.com/config#/|/pledge-duplicate

Things I've managed to accomplish:
1) Generally recreating the Google form in an HTML form
2) Integrating CSS/styling for a single page (form doesn't look perfect, but it's far from ugly)
3) Integrating inline Javascript.  Granted, all it does is pop up a window that says "Woah!", but from there we can do a lot.

What I haven't accomplished (yet):
1) Actually submitting this form info to a BeeSafe DB somewhere
2) Basic validation, but need to find a balance between frustrating restrictions and openness to pranks/bots
3) Any sort of GoogleMaps/MapBox integration to pre-check which region the submission might fit into

Please let me know if anything sticks out as user unfriendly in this draft.

**Comment by Beesafedavid** [comment]
Thanks, Zeb!
Looks like I don’t have access.
Screenshots, please.

**Comment by ZebGirouard** [comment]
![screenshot 2016-04-11 at 10 01 37 pm](https://cloud.githubusercontent.com/assets/5480464/14449499/00a21f48-0031-11e6-88ff-205d91d6b871.png)

and...

![screenshot 2016-04-11 at 10 02 08 pm](https://cloud.githubusercontent.com/assets/5480464/14449503/11d97324-0031-11e6-9991-53a2c1b4d219.png)

**Comment by ZebGirouard** [comment]
@Beesafedavid suggested a subtitle for all levels with actual definitions of Levels from BeeSafe standard.

**Comment by ZebGirouard** [comment]
@Beesafedavid , do you still think these subtitles are necessary when it says right above the pledge what the levels are?

![screenshot 2016-04-27 at 9 45 47 am](https://cloud.githubusercontent.com/assets/5480464/14858349/03c4c47c-0c5d-11e6-867e-23424fb59b55.png)

This can be tough to convey with screenshots, but I should point out that the only thing I'm changing from @kristinreynolds and @DaveLaskarzewski 's pledge page is the Google Form for pledge submissions.

**Comment by Beesafedavid** [comment]
On reflection, I have changed my mind. 
Subtitles are not necessary.
Thanks!

> On Apr 27, 2016, at 9:48 AM, Zeb Tim Girouard notifications@github.com wrote:
> 
> @Beesafedavid https://github.com/Beesafedavid , do you still think these subtitles are necessary when it says right above the pledge what the levels are?
> 
>  https://cloud.githubusercontent.com/assets/5480464/14858349/03c4c47c-0c5d-11e6-867e-23424fb59b55.png
> This can be tough to convey with screenshots, but I should point out that the only thing I'm changing from @kristinreynolds https://github.com/kristinreynolds and @DaveLaskarzewski https://github.com/DaveLaskarzewski 's pledge page is the Google Form for pledge submissions.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub https://github.com/BeeSafeEarth/BeeSafeEarth/issues/6#issuecomment-215127225

=== David Wheeler, Kansuke II ===
Musician (Shakuhachi) + Musicologist
3660 Buckeye Ct., Boulder, CO 80304 USA
shakudavid@gmail.com  
Cell: 720-545-5595
www.kansuke2.com http://www.kansuke2.com/

**Comment by ZebGirouard** [comment]
@boulderalf , do you know of an easy way to post to Google Sheets without Google Forms?  The bummer about Squarespace is that I'm limited to Vanilla JS.  I suppose Google Sheets API _should_ work, but wanted to know if you have experience before I go too far down that road.

**Comment by DaveLaskarzewski** [comment]
@Zeb, what do you mean by subtitles? 

Here comes a monkey wrench: I think that the verbiage around the levels should be swapped: 

Best! (Bee Friendly): I will not use....

Better (Organic): I will not use....

Good  (Neonic-free): I will not use....

@David, is this still an issue with the creator of this system? What I've listed above is easier to understand from a consumer standpoint.

**Comment by boulderalf** [comment]
@ZebGirouard I do not know of a supported way to do it.  But the Google Forms just do a POST to some web service.  If you turn on the developer tools, you can see what is being posted...

![image](https://cloud.githubusercontent.com/assets/2962724/14870683/20edb87e-0c99-11e6-84d2-3546aa63c648.png)

**Comment by ZebGirouard** [comment]
@boulderalf and @kristinreynolds , this post from HTML form to private Google Sheets appears to be _impossible_ without some form of user authentication...i.e. the owner of the sheet.  I believe that is not true if we're working on a site we own and can verify we own it before posting to Google Sheets.  Then we can (I think) use this fellow's article to do the heavy lifting in Google Apps:

https://mashe.hawksey.info/2011/10/google-spreadsheets-as-a-database-insert-with-apps-script-form-postget-submit-method/

That's the main reason the POST that Alfred posted above will work, i.e. origin = https://docs.google.com.  Anything coming from Google is clean, and anything coming from elsewhere is evil in Google's eyes.

And now that I've said all that, why don't we just get a real Database on Hostgator or AWS?  This seems way more trouble than it's worth.

**Comment by boulderalf** [comment]
@ZebGirouard I agree that we should get a real database.  The google sheets approach was just an expedient thing to do when I was working with only Boulder County data.

I'm going to continue the discussion at #8 

cc @kristinreynolds
